lnwallet/channel.go

```diff
@@ -1825,6 +1943,79 @@ func (lc *LightningChannel) restoreStateLogs(
 		}
 	}
 
+	// Restore dangling remote log updates that we haven't signed for yet.
+	err := lc.restorePendingRemoteUpdates(
+		danglingRemoteUpdates, localCommitment.height,
+	)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// restorePendingRemoteUpdates restores the remote log updates that we haven't
+// yet signed for.
+func (lc *LightningChannel) restorePendingRemoteUpdates(
+	danglingRemoteUpdates []channeldb.LogUpdate,
+	localCommitmentHeight uint64) error {
+
+	if len(danglingRemoteUpdates) == 0 {
+		lc.log.Debugf("No dangling remote updates to restore")
+		return nil
+	}
+
+	lc.log.Debugf("Restoring %v dangling remote updates",
+		len(danglingRemoteUpdates))
+
+	var (
+		lastHtlcIndex uint64
+		containsHtlc  bool
+	)
+	for _, logUpdate := range danglingRemoteUpdates {
+		logUpdate := logUpdate
+
+		payDesc := lc.remoteLogUpdateToPayDesc(
+			&logUpdate, lc.localUpdateLog, localCommitmentHeight,
+		)
+
+		// Insert the update into the log. The log update index doesn't
+		// need to be incremented (hence the restore calls), because its
+		// final value was properly persisted with the last local
+		// commitment update.
+		switch payDesc.EntryType {
+		case Add:
+			lc.remoteUpdateLog.restoreHtlc(payDesc)
+
+			// Track last htlc index for sanity checking.
+			lastHtlcIndex = payDesc.HtlcIndex
+			containsHtlc = true
+
+		case FeeUpdate:
+			lc.remoteUpdateLog.restoreUpdate(payDesc)
+
+		default:
+			lc.remoteUpdateLog.restoreUpdate(payDesc)
+
+			lc.localUpdateLog.markHtlcModified(payDesc.ParentIndex)
+		}
+	}
+
+	// Sanity checks to ensure that we've restored a valid update log.
+	lastRemoteUpdate := lc.remoteUpdateLog.Back().Value.(*PaymentDescriptor)
+	if lastRemoteUpdate.LogIndex != lc.remoteUpdateLog.logIndex-1 {
+		panic(fmt.Sprintf("log index mismatch: "+
```

Comment:
Yes, I also haven't fully made up my mind about these kind of decisions. In general, I prefer hard crashes. But knowing that no code is perfect, it can also cause a lot of inconvenience. 

I think the most dangerous option is to just log an error and continue. That potentially causes all kinds of fall out that may be difficult to link to the inconsistency introduced here.

I will modify this to returning an error. The channel won't be started, which should be a strong enough trigger to report the issue.